sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTest.java

```diff
@@ -332,4 +326,103 @@ public void beginTrainingInvalidModelStatus(HttpClient httpClient, FormRecognize
+        client = getFormTrainingClient(httpClient, serviceVersion);
+        beginTrainingUnlabeledRunner((trainingFilesUrl, useTrainingLabels) -> {
+            SyncPoller<OperationResult, CustomFormModel> trainingPoller =
+                client.beginTraining(trainingFilesUrl, useTrainingLabels);
+            trainingPoller.waitForCompletion();
+            validateCustomModelData(trainingPoller.getFinalResult(), false);
+        });
+    }
+
+    /**
+     * Verifies the result of the training operation for a valid labeled model Id and multi-page PDF training set Url.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void beginTrainingWithTrainingLabelsForMultiPagePDFTrainingSet(HttpClient httpClient,
+        FormRecognizerServiceVersion serviceVersion) {
+        client = getFormTrainingClient(httpClient, serviceVersion);
+        beginTrainingMultipageRunner(trainingFilesUrl -> {
+            SyncPoller<OperationResult, CustomFormModel> trainingPoller =
+                client.beginTraining(trainingFilesUrl, true);
+            trainingPoller.waitForCompletion();
+            validateCustomModelData(trainingPoller.getFinalResult(), true);
+        });
+    }
+
+    /**
+     * Verifies the result of the training operation for a valid unlabeled model Id and multi-page PDF training set Url.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void beginTrainingWithoutTrainingLabelsForMultiPagePDFTrainingSet(HttpClient httpClient,
+        FormRecognizerServiceVersion serviceVersion) {
+        client = getFormTrainingClient(httpClient, serviceVersion);
+        beginTrainingMultipageRunner(trainingFilesUrl -> {
+            SyncPoller<OperationResult, CustomFormModel> trainingPoller =
+                client.beginTraining(trainingFilesUrl, false);
+            trainingPoller.waitForCompletion();
+            validateCustomModelData(trainingPoller.getFinalResult(), false);
+        });
+    }
+
+    /**
+     * Verifies the result of the training operation for a valid labeled model Id and include subfolder training set
+     * Url.
+     */
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("com.azure.ai.formrecognizer.TestUtils#getTestParameters")
+    public void beginTrainingWithoutTrainingLabelsIncludeSubfolder(HttpClient httpClient,
+        FormRecognizerServiceVersion serviceVersion) {
+        client = getFormTrainingClient(httpClient, serviceVersion);
+        beginTrainingUnlabeledRunner((trainingFilesUrl, useTrainingLabels) -> {
+            SyncPoller<OperationResult, CustomFormModel> trainingPoller = client.beginTraining(trainingFilesUrl,
+                useTrainingLabels, new TrainingFileFilter().setIncludeSubFolders(true).setPrefix(SUBFOLDER), null);
```

Comment:
Is there a separate test for when only the prefix is set?

Reply:
added